src/docfx/config/GlobConfig.cs

```diff
@@ -41,5 +42,35 @@ public GlobConfig(string[] include, string[] exclude, T value, bool isGlob = tru
             Value = value;
             IsGlob = isGlob;
         }
+
+        public bool Match(string filePath)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(filePath));
+
+            if (Exclude.Any(e => MatchItem(filePath, e)))
+                return false;
+            if (Include.Any(i => MatchItem(filePath, i)))
+                return true;
+
+            return false;
+        }
+
+        private bool MatchItem(string filePath, string pattern)
+        {
+            if (pattern == null)
+                return false;
+
+            if (IsGlob)
+            {
+                var glob = new GlobMatcher(pattern);
+                return glob.Match(filePath);
+            }
+            else
+            {
+                return pattern.EndsWith('/') ?
```

Comment:
do we allow users to specify directories without trailing `/`?

Reply:
As filename is also supported here, `/` is needed to distinguish.

Reply:
can we look ask the file system if it is a directory or a file?

Reply:
Still not want to do this. It will be a bit confused when we support features like FAL. Another option is to drop support of file, which needs more investigation of existing usage...